consensus/service/src/tx_manager.rs

```diff
@@ -290,10 +290,25 @@ impl<E: ConsensusEnclave, UI: UntrustedInterfaces> TxManager<E, UI> {
         parent_block: &Block,
     ) -> TxManagerResult<(Block, BlockContents, BlockSignature)> {
         let cache = self.lock_cache();
-        let encrypted_txs_with_proofs = tx_hashes
+
+        let (entries, not_found) = tx_hashes
             .iter()
             .map(|tx_hash| {
-                let entry = cache.get(tx_hash).ok_or_else(|| TxManagerError::NotInCache(*tx_hash))?;
+                cache
+                    .get(tx_hash)
+                    .map_or_else(|| (*tx_hash, None), |entry| (*tx_hash, Some(entry)))
+            })
+            .partition::<Vec<_>, _>(|(_tx_hash, result)| result.is_some());
```

Comment:
Nice - this is a complicated enough construction that it may warrant a single line comment on L293, such as "Check cache for each tx_hash, and only form block if all hashes are in the cache"